modules/ding_seo/ding_seo.tokens.inc

```diff
@@ -0,0 +1,250 @@
+<?php
+
+/**
+ * Implements hook_token_info().
+ * 
+ * Custom tokens for JSON LD. 
+ * A new token type 'Ding Seo' is added to give easier access to the tokens from the UI.  
+ * Tokens is defined with this hook, content is added later with hook_tokens()
+ * 
+ */
+
+function ding_seo_token_info()
+{
+    $type = [
+        'name' => t('Ding Seo'),
+        'description' => t('Tokens for structured data as JSON LD.'),
+    ];
+
+    // Location tokens
+    $node['location_name'] = [
+        'name' => t("Location Name"),
+        'description' => t('The location is the '),
+    ];
+    $node['location_url'] = [
+        'name' => t("Location Url"),
+        'description' => t('location_url'),
+    ];
+    $node['postalCode'] = [
+        'name' => t("Postal Code"),
+        'description' => t('Postal Code'),
+    ];
+    $node['streetAddress'] = [
+        'name' => t("Street Address"),
+        'description' => t('Street Address'),
+    ];
+    $node['addressLocality'] = [
+        'name' => t("Address Locality"),
+        'description' => t('Address Locality'),
+    ];
+    $node['addressRegion'] = [
+        'name' => t("Address Region"),
+        'description' => t('Address Region'),
+    ];
+    $node['addressCountry'] = [
+        'name' => t("Address Country"),
+        'description' => t('Address Country'),
+    ];
+    $node['longitude'] = [
+        'name' => t("longitude"),
+        'description' => t('longitude'),
+    ];
+    $node['latitude'] = [
+        'name' => t("latitude"),
+        'description' => t('latitude'),
+    ];
+  
+    // Logo dimensions tokens
+    $node['logo_width'] = [
+        'name' => t("logo_width"),
+        'description' => t('logo_width'),
+    ];
+    $node['logo_height'] = [
+        'name' => t("logo_height"),
+        'description' => t('logo_height'),
+    ];
+
+    // Event data formatting
+    $node['event_startdate'] = [
+        'name' => t("Event Startdate"),
+        'description' => t('Event enddate in YYYY-MM-DD HH:MM::SS format'),
+    ];
+    $node['event_enddate'] = [
+        'name' => t("Event Enddate"),
+        'description' => t('Event enddate in YYYY-MM-DD HH:MM::SS format'),
+    ];
+    $node['event_free'] = [
+        'name' => t("Event Free Boolean"),
+        'description' => t('Boolean of whether the event is free or not '),
+    ];
+
+    // Language
+    $node['language'] = [
+        'name' => t("language"),
+        'description' => t('language'),
+    ];
+
+    // Returns the new token type as well as the tokens
+    return [
+        'types' => ['dingseotoken' => $type],
```

Comment:
I'm defining several ding_seo specific token types, so would it be possible to define your token type something like (assuming they are all event related):

```
  $info['types']['ding_seo_node_tokens'] = [
    'name' => t('Ding SEO node tokens'),
    'description' => t('Special node tokens defined by Ding SEO for use in structured data'),
    'needs-data' => 'node',
  ]; 
```
(also note the 'needs-data' option)

Otherwise I think it will be very confusing where to find the different tokens :)

Reply:
Another thing. If you look at my [`hook_token_info()` implementation](https://github.com/vejlebib/ding2/pull/7/files#diff-f6e47321f5e4c50d96d43b297bfbe4bfR17), I add the token types to the ting_object token and not as a "Global token".

I think it has the advantage of making it more clear that this token only works on the node token and the "needs-data" enforces this.

If you don't have "needs-data" wont these tokens just show up like you can use them everywhere, even when there's no node context?

Reply:
I've rewritten the token functions to use node & needs data.